File: python/pro2.py
# -*- coding: utf-8 -*-

def rec(st, nd, sum, limit):
    if (nd > limit):
        return sum
    else:
        if (0 == nd % 2):
            return rec(nd, st+nd, sum+nd, limit)
        else:
            return rec(nd, st+nd, sum, limit)

def pro2(limit):
    return rec(1, 2, 0, limit)

File: clojure/pro2.clj
(defn rec [st nd sum limit]
  (if (> nd limit)
    sum
    (rec nd (+ st nd)
         (if (= 0 (mod nd 2))
           (+ sum nd)
           sum)
         limit)))

(defn pro2 [limit]
  (rec 1 2 0 limit))